### PR TITLE
fix: timed out fetching a new connection from the connection pool.

### PR DESCRIPTION
**Pull request by youxq**
Fixes: prisma/prisma#25162

## solution
Change `new MakeTlsConnector`, which is executed by each `PostgreSql::new`, to obtain it from `MakeTlsConnectorManager.get_connector`. The logic of `get_connector` is to create `new MakeTlsConnector` in the first call and save the created tls connector. All subsequent calls will clone the tls connector from the first call, thereby reducing the overhead of the `new MakeTlsConnector`.
## test results
Use https://github.com/youxq/prisma-connection-test tool to test. 

The bug of `Timed out fetching a new connection from the connection pool.` was successfully triggered before the fix.
image
It no longer appears after the fix.
image

### CODE REVIEW

**Comment by rwest202** [comment]
Is this ready to merge?

**Comment by aqrln** [comment]
Thanks so much for the PR and sorry for the delay, I'll get to it soon (tomorrow or early next week).

**Comment by youxq** [comment]
I have fixed the lint issue. Please help trigger the checks workflow again. Thanks.

**Comment by michael-rzero** [comment]
Would love to see this merged

**Review by anton-ax — APPROVED** [review]

**Comment by youxq** [comment]
image
Can someone tell me how to fix this import issue in React Native compilation? Thanks

**Comment by semoal** [comment]
Seeing this error in production environment, even using pgbouncer, would be nice to ship this in the next release

**Comment by TheBubbleDuck** [comment]
Agreed with others! We are having similar issues as well and it would be great to see this make it into a release. Even if there is was an RC or beta to test it out to get more consensus and field testing if that would help community feel better about it.

**Comment by youxq** [comment]
@aqrln @anton-ax Hi, I have resolved the conflict. Please take a look. thanks

**Comment by tmm1** [comment]
@aqrln can we get some movement here

cc #5011

**Comment by michael-rzero** [comment]
Would love to see this released, would fix an issue we're seeing

**Comment by aqrln** [comment]
It's on my radar, sorry for the lack of movement so far! I approved the CI run for the latest changes.

**Comment by michael-rzero** [comment]
Awesome, thank you @aqrln - looks like all passed - thank you for keeping this on the go :)

**Comment by youxq** [comment]
Sorry, I just fixed another rustfmt issue, please help trigger the CI again, thanks @aqrln

**Comment by jkomyno** [comment]
> Sorry, I just fixed another rustfmt issue, please help trigger the CI again, thanks @aqrln

I have triggered it :)

**Comment by ajhollowayvrm** [comment]
Would LOVE to see this merged in as I've recently run into this issue while running Prisma in Kubernetes in GCP.

**Comment by aqrln** [comment]
Hey folks, I've triggered a dev release, you can test it by using Prisma version `5.22.0-integration-engines-5-22-0-22-gh-4998-integration-eb7519dcdf552de090269ebfa2beabc8e1a13e1b.1` (just make sure to install the exact version without `^`, otherwise you can get other random stuff from the `integration` dist-tag built from other branches).

**Comment by tmm1** [comment]
FWIW we've been running this in production for a few days now and it works as advertised.

File: quaint/src/pooled.rs [hunk 1]
```diff
@@ -314,7 +314,11 @@ impl Builder {
             url.set_flavour(flavour);
         }
 
-        if let QuaintManager::Postgres { ref mut url } = self.manager {
+        if let QuaintManager::Postgres {
+            ref mut url,
+            tls_manager: _,
+        } = self.manager
+        {
```
Comment (aqrln) [inline quaint/src/pooled.rs:321]: nit:

```suggestion
        if let QuaintManager::Postgres { ref mut url, .. } = self.manager {
```

**Review by aqrln — APPROVED** [review]
Looks great to me, thanks for doing this work and sorry for the delay in reviewing!

**Pull request merged**